x-pack/plugins/lens/public/visualizations/xy/visualization.tsx

```diff
@@ -944,6 +944,93 @@ export const getXyVisualization = ({
+        (groupId === 'left' && state.yLeftScale === 'log') ||
+        (groupId === 'right' && state.yRightScale === 'log')
+    );
+
+    if (logAxisGroups.length > 0) {
+      logAxisGroups
+        .map((axis) => {
+          const mixedDomainSeries = axis.series.filter((series) => {
+            let hasNegValues = false;
+            let hasPosValues = false;
+            const arr = activeData?.[series.layer]?.rows ?? [];
+            for (let index = 0; index < arr.length; index++) {
+              const value = arr[index][series.accessor];
+
+              if (value < 0) {
+                hasNegValues = true;
+              } else {
+                hasPosValues = true;
+              }
+
+              if (hasNegValues && hasPosValues) {
+                return true;
+              }
+            }
+
+            return false;
+          });
+          return {
+            ...axis,
+            mixedDomainSeries,
+          };
+        })
+        .forEach((axisGroup) => {
+          if (axisGroup.mixedDomainSeries.length === 0) return;
+          const { groupId } = axisGroup;
+
+          warnings.push({
+            uniqueId: `mixedLogScale-${groupId}`,
+            severity: 'warning',
+            shortMessage: '',
+            longMessage: (
+              <FormattedMessage
+                id="xpack.lens.xyVisualization.mixedLogScaleWarning"
+                defaultMessage="The {axisName} axis is set to a logarithmic scale but {n} of the datasets contains positive and negative data."
+                values={{
+                  axisName:
+                    groupId === 'left' ? (
+                      <FormattedMessage
+                        id="xpack.lens.xyVisualization.mixedLogScaleWarningLeft"
+                        defaultMessage="left"
+                      />
+                    ) : (
+                      <FormattedMessage
+                        id="xpack.lens.xyVisualization.mixedLogScaleWarningRight"
+                        defaultMessage="right"
+                      />
+                    ),
+                  n: axisGroup.mixedDomainSeries.length,
+                }}
+              />
+            ),
+            displayLocations: [{ id: 'toolbar' }],
+            fixableInEditor: true,
+          });
+
+          axisGroup.mixedDomainSeries.forEach(({ accessor }) => {
+            warnings.push({
+              uniqueId: `mixedLogScale-dimension-${accessor}`,
+              severity: 'warning',
+              shortMessage: '',
+              longMessage: (
+                <FormattedMessage
+                  id="xpack.lens.xyVisualization.mixedLogScaleWarning"
+                  defaultMessage="This metric is used on logarithmic scale and contains positive and negative data."
+                />
+              ),
+              displayLocations: [{ id: 'dimensionButton', dimensionId: accessor }],
+              fixableInEditor: true,
+            });
+          });
+        });
```

Comment:
Preview of these warnings shown below...

https://github.com/elastic/kibana/assets/19007109/3ade5b5a-05c0-4675-b19a-4bd7c44f1e03

Top metric on left axis, bottom 3 metrics on right axis.